### PR TITLE
Remove automatic agent unenrollment

### PR DESCRIPTION
**Pull request by hop-dev**
## What is the problem this PR solves?

https://github.com/elastic/fleet-server/issues/2063 Remove the unenrollment of agents after timeout. The `unenroll_timeout` field on agent policies will now be ignored, this has been replaced by a configurable inactivity timeout which does not unenroll the agent and is not managed by fleet server. 

I have added this as a breaking change as it removes functionality is that correct? 

## How does this PR solve the problem?

Removes any code and tests related to agent unenrollment

## How to test this PR locally

Enroll an agent using fleet server, note that the agent does not get unenrolled after `unenrollment_timeout`

## Design Checklist

- [x] I have ensured my design is stateless and will work when multiple fleet-server instances are behind a load balancer.
- [ ] (❌ I dont believe this is needed) I have or intend to scale test my changes, ensuring it will work reliably with 100K+ agents connected.
- [ ] (❌ I dont believe this is needed) I have included fail safe mechanisms to limit the load on fleet-server: rate limiting, circuit breakers, caching, load shedding, etc. 

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] (❌ I dont believe this is needed)I have made corresponding change to the default configuration files
- [ ] (❌ I dont believe this is needed)I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues
- Closes [#123](https://github.com/elastic/fleet-server/issues/2063)

### CODE REVIEW

File: internal/pkg/dl/agent.go [hunk 1]
```diff
@@ -67,28 +66,3 @@ func FindAgent(ctx context.Context, bulker bulk.Bulk, tmpl *dsl.Tmpl, name strin
 
 	return agent, nil
 }
-
-func FindOfflineAgents(ctx context.Context, bulker bulk.Bulk, policyID string, unenrollTimeout time.Duration, opt ...Option) ([]model.Agent, error) {
-	o := newOption(FleetAgents, opt...)
-	past := time.Now().UTC().Add(-unenrollTimeout).Format(time.RFC3339)
```
Comment (michel-laterman) [inline internal/pkg/dl/agent.go:73]: Does anything else use the `QueryOfflineAgentsByPolicyID` var/query?

File: internal/pkg/dl/agent.go [hunk 2]
```diff
@@ -67,28 +66,3 @@ func FindAgent(ctx context.Context, bulker bulk.Bulk, tmpl *dsl.Tmpl, name strin
...
-	res, err := Search(ctx, bulker, QueryOfflineAgentsByPolicyID, o.indexName, map[string]interface{}{
```
Comment (michel-laterman) [inline internal/pkg/dl/agent.go:74]: Does anything else use the `prepareOfflineAgentsByPolicyID` var/query?

Comment (hop-dev) [reply internal/pkg/dl/agent.go:74]: Good spot, I've removed it 👍

**Review by michel-laterman — APPROVED** [review]

**Pull request merged**